### ISSUE TITLE
Document the resize quirk for ConPTY

### ISSUE THREAD

**Issue by Unit2795** [microsoft/terminal#16911]
# Description of the new feature/enhancement

When I issue a [ResizePseudoConsole](https://learn.microsoft.com/en-us/windows/console/resizepseudoconsole) call, if the buffer is changing from a smaller size to a larger one, ConPTY outputs the reflowed buffer and overwrites any content currently in the terminal. If there were 50 lines of output, the buffer size was 24 lines, and I resize to 30 lines, the output will be overwritten with 24 lines. (The amount of content in the buffer at the time of the resize).

# Proposed technical implementation details

Possible Solutions:

1. Add the ability to disable the automatic output of the reflowed buffer
2. Add ability to specify differing buffer and window sizes for ConPTY, such that the window size will ensure VT sequences are emitted in sync with the terminal ConPTY may be connected to and therefore output looks visually correct. While also allowing a scrollback buffer to be maintained and reflowed as needed.
    - Example Pseudocode:

        ```
        CreatePseudoConsole(windowSize, bufferSize, hInput, hOutput, dwFlags, phPC)
        ```

# Related
- https://github.com/microsoft/terminal/issues/5800
- https://github.com/wez/wezterm/issues/1265
- https://github.com/microsoft/terminal/issues/14291

**Comment by zadjii-msft** [comment]
in lieu of a longer, more thorough response:

* https://github.com/microsoft/terminal/blob/5b8e731e5d64eafc1df5bd0467d4df82028fd13a/src/cascadia/TerminalConnection/ConptyConnection.cpp#L323
* https://github.com/microsoft/terminal/blob/5b8e731e5d64eafc1df5bd0467d4df82028fd13a/src/inc/conpty-static.h#L26
* https://github.com/microsoft/terminal/blob/5b8e731e5d64eafc1df5bd0467d4df82028fd13a/src/winconpty/winconpty.cpp#L140-L147

That's in the OpenConsole version of conpty. I forget if we shipped that to the OS conpty. We never really documented it cause it felt like a hack at the time, but we've been using for a few years now. Also I think we assumed that a caller who passed that would need to reflow their own buffer in the exact same way, to keep things in sync. 


But that might be helpful to experiment with?

**Comment by Unit2795** [comment]
I saw this flag mentioned in another issue (I can't recall which one). I gave it a try but it had no effect, it doesn't appear to be in the OS version. I'm not sure how to obtain the openconsole version and ship it with my library yet but I will research this. I saw the issue mentioning shipping ConPTY on nuget, I'm looking forward to that. Assuming I get the resize quirk working I will definitely have to do my own reflow. Having that functionality available would be great. If I need to extract it directly from this repo and put it up on nuget I'm not opposed (with proper credit).

Pardon my noobiness. Thanks for the reply.

**Comment by Unit2795** [comment]
For anyone who may be curious, I managed to obtain the OpenConsole.exe and ConPTY.dll prebuilt from this link (https://dev.azure.com/ms/terminal/_build?definitionId=136) by finding the latest build for the main branch, then found the Build x64 stage's build-x64-Release.zip artifact for the main branch.

![2024-03-25_02-07-35](https://github.com/microsoft/terminal/assets/25918877/18ffadba-0922-4ff3-af19-17b60aaee1d3)
![2024-03-25_02-07-57](https://github.com/microsoft/terminal/assets/25918877/866409e4-53d2-4514-985f-ba8ceb43ec8d)
![2024-03-25_02-08-56](https://github.com/microsoft/terminal/assets/25918877/532fc457-2e98-4353-9264-7926d938effe)

**Comment by Unit2795** [comment]
After downloading the latest ConPTY.dll, PSEUDOCONSOLE_RESIZE_QUIRK did work. Thank you for pointing this out @zadjii-msft . I've since adjusted this issue to remove the request for disabling reflow output. I did add another request for a possible plain-text render feature as it seemed like it could be related to my request for programmatic reflow. If I need to break that out into another issue I can do that.
